File: ABC/ABC301-400/ABC338/E/fenwicktree/E.py
from atcoder.fenwicktree import FenwickTree

N = int(input())
P = [[0, 0] for _ in range(N)]
fwA = FenwickTree(2 * N)
fwB = FenwickTree(2 * N)
for i in range(N):
	P[i] = list(map(int, input().split()))
	P[i][0] -= 1
	P[i][1] -= 1
	if P[i][0] > P[i][1]:
		P[i][0], P[i][1] = P[i][1], P[i][0]
	fwA.add(P[i][0], 1)
	fwB.add(P[i][1], 1)
P.sort()
for A, B in P:
	if fwB.sum(A + 1, B) < fwA.sum(A + 1, B):
		print("Yes")
		exit()
	fwA.add(A, -1)
	fwB.add(B, -1)
print("No")

File: ABC/ABC301-400/ABC338/E/fenwicktree/E.cpp
#include <bits/stdc++.h>
#include <atcoder/fenwicktree>
using namespace std;
using namespace atcoder;

int main() {
	int N;
	cin >> N;
	vector<int> A(N), B(N);
	fenwick_tree<int> fwA(2 * N), fwB(2 * N);
	for (int i = 0; i < N; i++) {
		cin >> A[i] >> B[i];
		A[i]--; B[i]--;
		if (A[i] > B[i]) {
			swap(A[i], B[i]);
		}
		fwA.add(A[i], 1);
		fwB.add(B[i], 1);
	}
	for (int i = 0; i < N; i++) {
		if (fwB.sum(A[i] + 1, B[i]) < fwA.sum(A[i] + 1, B[i])) {
			cout << "Yes\n";
			return 0;
		}
	}
	cout << "No\n";
}